src/access.rs

```diff
@@ -0,0 +1,365 @@
+		}
+
+		// Check all registered outputs for dependent spending transactions.
+		let registered_outputs: Vec<WatchedOutput> = {
+			let locked_queued_outputs = self.queued_outputs.lock().unwrap();
+			let locked_watched_outputs = self.watched_outputs.lock().unwrap();
+			locked_watched_outputs.iter().chain(locked_queued_outputs.iter()).cloned().collect()
+		};
+
+		// Remember all registered outputs that haven't been spent for future processing.
+		let mut unspent_registered_outputs = Vec::new();
+
+		for output in registered_outputs {
+			if let Some(output_status) = client
+				.get_output_status(&output.outpoint.txid, output.outpoint.index as u64)
+				.await?
+			{
+				if output_status.spent {
+					if let Some(spending_tx_status) = output_status.status {
+						if spending_tx_status.confirmed {
+							let spending_txid = output_status.txid.unwrap();
+							if let Some(spending_tx) = client.get_tx(&spending_txid).await? {
+								let block_height = spending_tx_status.block_height.unwrap();
+								// TODO: Switch to `get_header_by_hash` once released upstream (https://github.com/bitcoindevkit/rust-esplora-client/pull/17)
+								let block_header = client.get_header(block_height).await?;
+								if let Some(merkle_proof) =
+									client.get_merkle_proof(&spending_txid).await?
+								{
+									confirmed_txs.push((
+										spending_tx,
+										block_height,
+										block_header,
+										merkle_proof.pos,
+									));
+									continue;
+								}
+							}
+						}
+					}
+				}
+			}
+			unspent_registered_outputs.push(output);
+		}
+
+		// Sort all confirmed transactions first by block height, then by in-block
+		// position, and finally feed them to the interface in order.
+		confirmed_txs.sort_unstable_by(
+			|(_, block_height1, _, pos1), (_, block_height2, _, pos2)| {
+				block_height1.cmp(&block_height2).then_with(|| pos1.cmp(&pos2))
+			},
+		);
+		for (tx, block_height, block_header, pos) in confirmed_txs {
+			for c in confirmables {
+				c.transactions_confirmed(&block_header, &[(pos, &tx)], block_height);
+			}
+		}
+
+		*self.queued_transactions.lock().unwrap() = Vec::new();
+		*self.watched_transactions.lock().unwrap() = unconfirmed_registered_txs;
+		*self.queued_outputs.lock().unwrap() = Vec::new();
+		*self.watched_outputs.lock().unwrap() = unspent_registered_outputs;
+
+		Ok(())
+	}
+
+	async fn sync_transaction_unconfirmed(
+		&self, confirmables: &Vec<&(dyn Confirm + Sync)>,
+	) -> Result<(), Error> {
+		let client = &*self.blockchain;
+		// Query the interface for relevant txids and check whether they have been
+		// reorged-out of the chain.
+		let relevant_txids =
+			confirmables.iter().flat_map(|c| c.get_relevant_txids()).collect::<HashSet<Txid>>();
+		for txid in relevant_txids {
+			let tx_unconfirmed = client
+				.get_tx_status(&txid)
+				.await
+				.ok()
+				.unwrap_or(None)
+				.map_or(true, |status| !status.confirmed);
+			if tx_unconfirmed {
```

Comment:
IIUC the docs correctly, I think we don't, as long as it  still confirmed?

Reply:
Hmpf, good point. How about we change `Confirm::get_relevant_txids()` to return `Vec<(Txid, BlockHash)>` (i.e. `Channel::funding_tx_confirmed_in`), which would allow us here (and users general) to check that the relevant transactions are still confirmed in the specific given block?

Reply:
IIUC, tracking the block hash in which a transaction was confirmed would also relax the last ordering requirement "Unconfirmed transactions must be given after the original confirmations and before any reconfirmation.", as then only previously-confirmed `Txid`s could be returned by `get_relevant_txids()` anyways?

Reply:
Draft for now: https://github.com/lightningdevkit/rust-lightning/pull/1796